adapters/sidekiq/test/opentelemetry/adapters/sidekiq/adapter_test.rb

```diff
@@ -0,0 +1,52 @@
+# frozen_string_literal: true
+
+# Copyright 2020 OpenTelemetry Authors
+#
+# SPDX-License-Identifier: Apache-2.0
+
+require 'test_helper'
+require_relative '../../../../lib/opentelemetry/adapters/sidekiq'
+
+class SimpleEnqueueingJob
+  include Sidekiq::Worker
+
+  def perform
+    SimpleJob.perform_async
+  end
+end
+
+class SimpleJob
+  include Sidekiq::Worker
+
+  def perform
+    puts 'Simple work accomplished'
+  end
+end
+
+describe OpenTelemetry::Adapters::Sidekiq::Adapter do
+  let(:adapter) { OpenTelemetry::Adapters::Sidekiq::Adapter.instance }
+  let(:exporter) { EXPORTER }
+  before { exporter.reset }
+
+  describe 'tracing' do
+    before do
+      adapter.install
+    end
+
+    it 'before performing any jobs' do
+      _(exporter.finished_spans.size).must_equal 0
+    end
+
+    it 'after performing a simple job' do
+      Sidekiq::Testing.inline! { SimpleJob.perform_async }
+
+      _(exporter.finished_spans.size).must_equal 2
+    end
+
+    it 'after performing a simple job enqueuer' do
+      Sidekiq::Testing.inline! { SimpleEnqueueingJob.perform_async }
+
+      _(exporter.finished_spans.size).must_equal 4
```

Comment:
So right now this spec generates four spans.

1) SimpleEnqueueingJob gets enqueued and the client middleware creates a span.
2) SimpleEnqueueingJob gets run by the server and the server middleware creates a span.
3) SimpleJob gets enqueued and the client middleware creates a span.
4) SimpleJob gets run by the server and the server middleware creates a span.

Currently the code is not connecting the first step with the other three. So only of 3/4 spans produced in this test share the same trace_id.  

As I understand it, this should be configurable so that the three subsequent spans are connected by a link or by a trace_id.

Reply:
2 should default to the same trace_id as 1, but allow configuring to start a new trace with a link to 1.
3 should have the same trace_id as 2.
4 should default to the same trace_id as 3, but allow configuring to start a new trace with a link to 3.